pkg/asset/machines/aws/machines.go

```diff
@@ -0,0 +1,120 @@
+// Package aws generates Machine objects for aws.
+package aws
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/utils/pointer"
+	awsprovider "sigs.k8s.io/cluster-api-provider-aws/pkg/apis/awsproviderconfig/v1alpha1"
+	clusterapi "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+
+	"github.com/openshift/installer/pkg/types"
+)
+
+// Machines returns a list of machines for a machinepool.
+func Machines(config *types.InstallConfig, pool *types.MachinePool, role, userDataSecret string) ([]clusterapi.Machine, error) {
+	if configPlatform := config.Platform.Name(); configPlatform != types.PlatformNameAWS {
+		return nil, fmt.Errorf("non-AWS configuration: %q", configPlatform)
+	}
+	if poolPlatform := pool.Platform.Name(); poolPlatform != types.PlatformNameAWS {
+		return nil, fmt.Errorf("non-AWS machine-pool: %q", poolPlatform)
+	}
+	clustername := config.ObjectMeta.Name
+	platform := config.Platform.AWS
+	mpool := pool.Platform.AWS
+	azs := mpool.Zones
+
+	total := int64(1)
+	if pool.Replicas != nil {
+		total = *pool.Replicas
+	}
+	numOfAZs := int64(len(azs))
+	var machines []clusterapi.Machine
+	for idx, az := range azs {
+		replicas := int32(total / numOfAZs)
+		if int64(idx) < total%numOfAZs {
+			replicas++
+		}
+
+		tags, err := tagsFromUserTags(config.ClusterID, clustername, platform.UserTags)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to create awsprovider.TagSpecifications from UserTags")
+		}
+		provider := awsprovider.AWSMachineProviderConfig{
```

Comment:
It looks like this can be factored out into a shared helper (like you've already factored out `tagsFromUserTags`).

Reply:
done